fe/src/main/java/org/apache/doris/analysis/CastExpr.java

```diff
@@ -219,7 +195,9 @@ private void analyze() throws AnalysisException {
 
     @Override
     public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
-        type = targetType;
+        Preconditions.checkState(!isImplicit);
+        targetTypeDef.analyze(analyzer);
```

Comment:
targetTypeDef will null if this is implicit cast? And this will throw a NullPointerException

Reply:
Implicit CastExpr can't call analyze() outside , so targetTypeDef will not throw NullPointerException.